src/types/entities/schelling.ts

```diff
@@ -138,13 +212,24 @@ export class SchellingGame {
 	// --- claim
 	public claim(
 		winner: Reveal,
+		owner: string,
 		amount: BigNumber,
 		block: BlockDetails,
 		freezes: StakeFreeze[] = [],
 		slashes: StakeSlash[] = []
 	) {
 		const roundNo = SchellingGame.roundFromBlockNo(block.blockNo)
 
+		const round = this.getOrCreateRound(roundNo, block)
+
+		// create the player if it doesn't exist
+		if (!this.players.has(winner.overlay)) {
```

Comment:
Seems like all 3 (commit, reveal, claim) start with both getOrCreateRound and creating the player if it doesn't exist.   Perhaps a new method that covers both of these and returns both the round and Player objects for use by the transactions?